### PR TITLE
update tendermint codec.proto and generated.rs

### PR DESCRIPTION
**Pull request by Seb-figment**

### CODE REVIEW

**Comment by kruspy** [comment]
Tests are failing for various reasons (missing Asc traits, etc.) and style checks are not passing (you can run `cargo fmt --all -- --check` before creating the PR and it will show you where you need to fix the style). Would need to fix that in order to merge.

File: chain/tendermint/src/runtime/abi.rs [hunk 1]
```diff
@@ -39,19 +39,6 @@ impl ToAscObj<AscEventData> for codec::EventData {
     }
 }
 
-impl ToAscObj<AscTransactionData> for codec::TransactionData {
```
[the remark this answers is not in the document]
Comment (Seb-figment) [reply chain/tendermint/src/runtime/abi.rs:42]: Just moved around. It is at line 907

File: chain/tendermint/src/runtime/abi.rs [hunk 2]
```diff
@@ -187,17 +173,17 @@ impl ToAscObj<AscConsensus> for codec::Consensus {
...
-        &self,
-        heap: &mut H,
-        gas: &GasCounter,
-    ) -> Result<AscData, DeterministicHostError> {
-        Ok(AscData {
-            txs: asc_new(heap, &self.txs, gas)?,
-        })
-    }
-}
+// impl ToAscObj<AscData> for codec::Data {
```
[the remark this answers is not in the document]
Comment (Seb-figment) [reply chain/tendermint/src/runtime/abi.rs:176]: done

File: chain/tendermint/src/runtime/abi.rs [hunk 3]
```diff
@@ -700,13 +686,225 @@ impl ToAscObj<AscTxResult> for codec::TxResult {
         Ok(AscTxResult {
             height: self.height,
             index: self.index,
-            tx: asc_new(heap, &Bytes(&self.tx), gas)?,
+            tx: asc_new_or_null(heap, &self.tx, gas)?,
```
[the remark this answers is not in the document]
Comment (Seb-figment) [reply chain/tendermint/src/runtime/abi.rs:689]: In proto3 spec all field are optional. Also with Firehose filtering we may want to send Block without transaction. I'll reach out in slack to get opinion of the rest of the team.

File: chain/tendermint/src/protobuf/fig.tendermint.codec.v1.rs [hunk 4]
```diff
@@ -407,7 +802,6 @@ pub enum SignedMsgType {
     /// Proposals
     Proposal = 32,
 }
-/// BlockIdFlag indicates which BlockID the signature is for
```
[the remark this answers is not in the document]
Comment (Seb-figment) [reply chain/tendermint/src/protobuf/fig.tendermint.codec.v1.rs:410]: This is cause by automation in tendermint-protobuf-def repo when files are merged together.

**Pull request closed without merging**